gusto/equations/shallow_water_equations.py

```diff
@@ -456,3 +654,11 @@ def __init__(self, domain, parameters, fexpr=None,
 
         # Use the underlying routine to do a first linearisation of the equations
         self.linearise_equation_set()
+
+
+def compute_saturation(q0, nu, H, g, D, b, B=None):
```

Comment:
Should we make this a method of the `ThermalShallowWaterEquations`? It could return `None` is there are no active tracers. I don't know if it would be better to be have a mixed field `X` as the argument or the individual fields, but this would also give access to the parameters and `B`

Reply:
@nhartney in your branch this is in its own special file all by itself - I was wondering why that was? Is it used anywhere else?

Reply:
There wasn't a good reason to have it in a file on its own - I did it only because I didn't know where it should belong. I also use the function elsewhere to set up some of the examples.